api/v1alpha2/virtualmachine_network_types.go

```diff
@@ -209,9 +209,13 @@ type VirtualMachineNetworkSpec struct {
 	// If the Interfaces field is empty and the Disabled field is false, then
 	// a default interface with the name eth0 will be created.
 	//
+	// The maximum number of network interface allowed is 10 because of the
```

Comment:
Please change to "because of the limit built into vSphere." That way it covers both vC and ESX.